--- a/src/alpm_config.vala	2023-07-11 14:48:54.325153600 +0200
+++ b/src/alpm_config.vala	2023-07-11 14:51:50.681800116 +0200
@@ -132,11 +132,13 @@ internal class AlpmConfig {
 			try {
 				var file = GLib.File.new_for_path (tmp_path);
 				if (!file.query_exists ()) {
+					Process.spawn_command_line_sync ("umask 077");
 					Process.spawn_command_line_sync ("mkdir -p %s".printf (tmp_path));
 					Process.spawn_command_line_sync ("chmod a+w %s".printf (tmp_path));
 				}
 				file = GLib.File.new_for_path (tmp_dbpath);
 				if (!file.query_exists ()) {
+					Process.spawn_command_line_sync ("umask 077");
 					Process.spawn_command_line_sync ("mkdir -p %s".printf (tmp_dbpath));
 					Process.spawn_command_line_sync ("chmod a+w %s".printf (tmp_dbpath));
 					Process.spawn_command_line_sync ("ln -s %slocal %s".printf (dbpath, tmp_dbpath));
